### PR TITLE
Cleanup and update

### PR DESCRIPTION
**Pull request by electrocucaracha**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR removes some unnecessary code and references as well as upgrades some dependencies used by integration tests and provisioning.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: e2e/provision/init.sh [hunk 1]
```diff
@@ -27,6 +27,11 @@ function get_status {
         printf "Memory free(Kb):"
         awk -v low="$(grep low /proc/zoneinfo | awk '{k+=$2}END{print k}')" '{a[$1]=$2}  END{ print a["MemFree:"]+a["Active(file):"]+a["Inactive(file):"]+a["SReclaimable:"]-(12*low);}' /proc/meminfo
     fi
+    if command -v docker >/dev/null; then
+        echo "Docker statistics:"
+        docker stats --no-stream
```
Comment (arora-sagar) [inline e2e/provision/init.sh:32]: @electrocucaracha What is the reason of using it here? 
It means we will have the stats only for a particular moment but not when it keeps on refreshing. Probably it is better to collect every couple of seconds using a cronjob? What do you think?

File: e2e/provision/playbooks/roles/bootstrap/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ Available variables are listed below, along with default values (see defaults/ma
 | gtp5g_version                   | no       | v0.6.8                                                        |                           | GTP5G source code version                                                         |
 | gtp5g_tarball_url               | no       |                                                               |                           | GTP5G tarball URI                                                                 |
 | k8s.context                     | no       | kind-kind                                                     |                           | Kubernetes context to create gitea resources                                      |
-| k8s.version                     | no       | v1.27.1                                                       |                           | Kubernetes version used for the management cluster                                |
+| k8s.version                     | no       | v1.29.2                                                       |                           | Kubernetes version used for the management cluster                                |
```
Comment (arora-sagar) [inline e2e/provision/playbooks/roles/bootstrap/README.md:26]: @electrocucaracha So we will have the same for management cluster and workload clusters right?

Comment (electrocucaracha) [reply e2e/provision/playbooks/roles/bootstrap/README.md:26]: No, in other to achieve that we need to modify the [workload cluster package](https://github.com/nephio-project/catalog/blob/main/infra/capi/cluster-capi-kind/cluster.yaml#L33)

**Comment by vjayaramrh** [comment]
@electrocucaracha will the failed jobs be root caused and fixed, holding on providing approval until the jobs pass

**Comment by electrocucaracha** [comment]
I'll ignore this error given the [flakiness test results](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/254/e2e-free5gc-ubuntu-focal/1777485098451472384#1:build-log.txt%3A73976)

/override

**Comment by electrocucaracha** [comment]
I'll ignore this error given the [flakiness test results](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/254/e2e-free5gc-ubuntu-focal/1777485098451472384#1:build-log.txt%3A73976)

/override e2e-free5gc-ubuntu-focal

**Pull request merged**